extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevServicesProcessor.java

```diff
@@ -363,20 +385,29 @@ public StartResult(String url, boolean createDefaultRealm, String realmNameToUse
         private boolean realmFileExists;
         private String hostName = null;
         private String realmNameToUse;
-        private boolean keycloakX;
+        private final boolean keycloakX;
 
         public QuarkusOidcContainer(DockerImageName dockerImageName, OptionalInt fixedExposedPort, boolean useSharedNetwork,
                 Optional<String> configuredRealmName, Optional<String> realmPath, String containerLabelValue,
                 boolean sharedContainer, Optional<String> javaOpts) {
             super(dockerImageName);
-            this.fixedExposedPort = fixedExposedPort;
+
             this.useSharedNetwork = useSharedNetwork;
             this.configuredRealmName = configuredRealmName;
             this.realmPath = realmPath;
             this.containerLabelValue = containerLabelValue;
             this.sharedContainer = sharedContainer;
             this.javaOpts = javaOpts;
             this.keycloakX = isKeycloakX(dockerImageName);
+
+            if (sharedContainer && fixedExposedPort.isEmpty()) {
```

Comment:
It looks fine; I've been trying to figure out if it can be a problem but if I understand it correctly effectively you are doing what test containers do if no fixed port is set.

Reply:
Exactly